Makefile

```diff
@@ -214,19 +214,25 @@ local-deploy-namespaced: docker-build kind-load-image ## Deploy the dns operator
 ##@ Build
 
 .PHONY: build
+build: GIT_SHA=$(shell git rev-parse HEAD || echo "unknown") 
+build: DIRTY=$(shell hack/check-git-dirty.sh || echo "unknown")
 build: manifests generate fmt vet ## Build manager binary.
-	go build -o bin/manager cmd/main.go
+	go build -ldflags "-X main.gitSHA=${GIT_SHA} -X main.dirty=${DIRTY}" -o bin/manager cmd/main.go
 
 .PHONY: run
+run: GIT_SHA=$(shell git rev-parse HEAD || echo "unknown") 
+run: DIRTY=$(shell hack/check-git-dirty.sh || echo "unknown")
 run: manifests generate fmt vet ## Run a controller from your host.
-	go run ./cmd/main.go --zap-devel --provider inmemory,aws,google,azure
+	go run -ldflags "-X main.gitSHA=${GIT_SHA} -X main.dirty=${DIRTY}" ./cmd/main.go --zap-devel --provider inmemory,aws,google,azure
```

Comment:
nit: Personally I'd prefer it if these weren't required at runtime. Seems to me default values could be set in code for these.

Reply:
It makes it more difficult to have them hardcoded as it requires using Git in Go. However, no problem with doing this if that is your preference. :)